K2Bridge/Factories/AggregateFactory.cs

```diff
@@ -28,17 +28,18 @@ internal static class AggregateFactory
         /// </summary>
         /// <param name="key">The aggregation key.</param>
         /// <param name="dataTable">The row collection be parsed.</param>
+        /// <param name="query">QueryData containing query information.</param>
         /// <param name="logger">ILogger object for logging.</param>
         /// <returns><see cref="BucketAggregate"/>.</returns>
-        public static BucketAggregate GetDateHistogramAggregate(string key, DataTable dataTable, ILogger logger)
+        public static BucketAggregate GetDateHistogramAggregate(string key, DataTable dataTable, QueryData query, ILogger logger)
```

Comment:
I see that you pass QueryData but use just one parameter from it.
Why not pass just that parameter?

Reply:
In the current implementation, TopHits returns only field value passed in the aggregation (and it works perfectly fine). However, it seems if fields are also specified at the root level, they should be reported also in the values returned. The only way to access these additional fields is to access QueryData. So I suggest to keep it as it for next iteration. I will add this information to the issue list.

``` json
"aggs": {
    "2": {
      "top_hits": {
        "docvalue_fields": [
          {
            "field": "products.base_price"
          }
        ],
        "_source": "products.base_price",
        "size": 1,
        "sort": [
          {
            "order_date": {
              "order": "asc"
            }
          }
        ]
      }
    }
  },
  "size": 0,
  "fields": [
    {
      "field": "customer_birth_date",
      "format": "date_time"
    },
    {
      "field": "order_date",
      "format": "date_time"
    },
    {
      "field": "products.created_on",
      "format": "date_time"
    }
  ],
```